targets/HUDL/main.cpp

```diff
@@ -13,53 +20,180 @@
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// CANopen specific Callbacks. Need to be defined in some location
+///////////////////////////////////////////////////////////////////////////////
+extern "C" void CONodeFatalError(void) {}
+
+extern "C" void COIfCanReceive(CO_IF_FRM* frm) {}
+
+extern "C" int16_t COLssStore(uint32_t baudrate, uint8_t nodeId) { return 0; }
+
+extern "C" int16_t COLssLoad(uint32_t* baudrate, uint8_t* nodeId) { return 0; }
+
+extern "C" void CONmtModeChange(CO_NMT* nmt, CO_MODE mode) {}
+
+extern "C" void CONmtHbConsEvent(CO_NMT* nmt, uint8_t nodeId) {}
+
+extern "C" void CONmtHbConsChange(CO_NMT* nmt, uint8_t nodeId, CO_MODE mode) {}
+
+extern "C" int16_t COParaDefault(CO_PARA* pg) { return 0; }
+
+extern "C" void COPdoTransmit(CO_IF_FRM* frm) {}
+
+extern "C" int16_t COPdoReceive(CO_IF_FRM* frm) { return 0; }
+
+extern "C" void COPdoSyncUpdate(CO_RPDO* pdo) {}
+
+extern "C" void COTmrLock(void) {}
+
+extern "C" void COTmrUnlock(void) {}
+
 int main() {
-    IO::GPIO* devices[deviceCount];
-    auto& reg_select = IO::getGPIO<IO::Pin::PA_3>(IO::GPIO::Direction::OUTPUT);
-    reg_select.writePin(IO::GPIO::State::LOW);
+    // Initialize system
+    IO::init();
 
-    auto& reset = IO::getGPIO<IO::Pin::PB_3>(IO::GPIO::Direction::OUTPUT);
-    reset.writePin(IO::GPIO::State::LOW);
-    time::wait(100);
-    reset.writePin(IO::GPIO::State::HIGH);
-    time::wait(100);
+    // Will store CANopen messages that will be populated by the EVT-core CAN
+    // interrupt
+    EVT::core::types::FixedQueue<CANOPEN_QUEUE_SIZE, IO::CANMessage> canOpenQueue;
 
-    auto& cs = IO::getGPIO<IO::Pin::PB_12>(IO::GPIO::Direction::OUTPUT);
-    cs.writePin(IO::GPIO::State::HIGH);
+    // Initialize CAN, add an IRQ which will add messages to the queue above
+    IO::CAN& can = IO::getCAN<IO::Pin::PA_12, IO::Pin::PA_11>();
+    can.addIRQHandler(canInterrupt, reinterpret_cast<void*>(&canOpenQueue));
 
-    auto& hudl_spi = IO::getSPI<IO::Pin::PB_13, IO::Pin::PB_15, IO::Pin::PC_11>(
+    // Initialize the timer
+    DEV::Timerf302x8 timer(TIM2, 100);
+    timer.stopTimer();
+
+    //create the RPDO node
+    IO::GPIO* devices[deviceCount];
+
+    IO::GPIO& regSelect = IO::getGPIO<IO::Pin::PA_3>(EVT::core::IO::GPIO::Direction::OUTPUT);
+    IO::GPIO& reset = IO::getGPIO<IO::Pin::PB_3>(EVT::core::IO::GPIO::Direction::OUTPUT);
+    devices[0] = &IO::getGPIO<IO::Pin::PB_12>(EVT::core::IO::GPIO::Direction::OUTPUT);
+    devices[0]->writePin(IO::GPIO::State::HIGH);
+    auto& hudl_spi = IO::getSPI<IO::Pin::SPI_SCK, IO::Pin::SPI_MOSI>(
         devices, deviceCount);
-    hudl_spi.configureSPI(SPI_SPEED, SPI_MODE3, SPI_MSB_FIRST);
 
-    auto board = HUDL::HUDL(reg_select, reset, cs, hudl_spi);
+    hudl_spi.configureSPI(SPI_SPEED, SPI_MODE0, SPI_MSB_FIRST);
 
-    // Setup UART
-    IO::UART& uart = IO::getUART<IO::Pin::UART_TX, IO::Pin::UART_RX>(9600);
+    auto hudl = HUDL::HUDL(regSelect, reset, hudl_spi);
```

Comment:
NIT: I'm not sure about this at all, but this declaration looks a little Java-ey to me. I think you could accomplish the same thing in a more standard C++ way with `HUDL::HUDL hudl(regSelect, reset, hudl_spi);`